addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/discovery/FreeboxDiscoveryService.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.freebox.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.matmaul.freeboxos.FreeboxException;
+import org.matmaul.freeboxos.FreeboxOsClient;
+import org.matmaul.freeboxos.login.Authorize;
+import org.matmaul.freeboxos.login.LoginManager;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.freebox.FreeboxBindingConstants.*;
+
+/**
+ * The {@link FreeboxDiscoveryService} is responsible for discovering a Freebox Thing
+ * on the local network
+ * 
+ * @author Gaël L'hopital - Initial contribution
+ */
+
+public class FreeboxDiscoveryService extends AbstractDiscoveryService {
+
+	private static final Logger logger = LoggerFactory.getLogger(FreeboxDiscoveryService.class);
+
+	public FreeboxDiscoveryService() {
+		super(SUPPORTED_THING_TYPES_UIDS,10);
+	}
+
+	public Set<ThingTypeUID> getSupportedThingTypes() {
+		return SUPPORTED_THING_TYPES_UIDS;
+	}
+	
+	@Override
+	protected void startScan() {
+		discoverFreebox();
+	}
+
+
+	/**
+	 * Discover the Freebox, get its appToken then submits the result
+	 */
+	private void discoverFreebox(){
+		Bundle bundle = FrameworkUtil.getBundle(getClass());			
+		
+		FreeboxOsClient fbClient = new FreeboxOsClient(
+					bundle.getSymbolicName() /*  org.openhab.binding.freebox */
+					);
+		
+		LoginManager loginManager = fbClient.getLoginManager();
+		
+		try {
+			Authorize authorize = loginManager.newAuthorize(
```

Comment:
When the discovery service is launched for the Freebox, a token is generated and the user has to validate it directly on the Freebox appliance. I'm no longer ready to test this again (sure it works) as currently PaperUI is broken